src/middleware/timing/timing.ts

```diff
@@ -221,3 +221,40 @@ export const endTime = (c: Context, name: string, precision?: number) => {
   setMetric(c, name, duration, description, precision)
   metrics.timers.delete(name)
 }
+
+/**
+ * Wrap a Promise to capture its duration.
+ * @param {Context} c - The context of the request.
+ * @param {string} name - The name of the timer.
+ * @param {Promise<T>} callable - The Promise to time.
+ * @param {string} [description] - The description of the timer.
+ * @param {number} [precision] - The precision of the timer value.
+ *
+ * @example
+ * ```ts
+ *   // Instead of this:
+ *   const data = await db.findMany(...);
+ *
+ *   // do this:
+ *   const data = await wrapTime(c, 'query', db.findMany(...));
+ * ```
+ * */
+export async function wrapTime<T>(
+  c: Context,
+  name: string,
+  callable: Promise<T>,
+  description?: string,
+  precision?: number
+): Promise<T> {
+  startTime(c, name, description)
+  let result
+  try {
+    result = await callable
```

Comment:
I think you can write like this for this line:

```ts
return await callable
```

and, the `catch() {}` is really necessary?

If not, you can apply this patch:

```diff
diff --git a/src/middleware/timing/timing.ts b/src/middleware/timing/timing.ts
index 4c81285c..02fa87bb 100644
--- a/src/middleware/timing/timing.ts
+++ b/src/middleware/timing/timing.ts
@@ -247,13 +247,8 @@ export async function wrapTime<T>(
   precision?: number
 ): Promise<T> {
   startTime(c, name, description)
-  let result
   try {
-    result = await callable
-    return result
-    // eslint-disable-next-line no-useless-catch
-  } catch (e) {
-    throw e
+    return await callable
   } finally {
     endTime(c, name, precision)
   }
```